blades/sub_blade.h

```diff
@@ -311,6 +315,58 @@ class BladeBase* SubBladeZZ(int first_led, int last_led, int stride, int column,
   return ret;
 }
 
+class SubBladeWrapperWithList : public SubBladeWrapper {
+public:
+  SubBladeWrapperWithList(const int* indices, int count)
+      : indices_(indices), count_(count) {}
+  void set(int led, Color16 c) override {
+    if (led >= 0 && led < count_) {
+      blade_->set(indices_[led], c);
+    }
+  }
+  void set_overdrive(int led, Color16 c) override {
+    if (led >= 0 && led < count_) {
+      blade_->set_overdrive(indices_[led], c);
+    }
+  }
+
+  int num_leds() const override { return count_; }
+
+protected:
+  const int* indices_;
+  int count_;
+};
+
+BladeBase* SubBladeWithList(const int* indices, int count, BladeBase* blade) {
+  if (blade) {
+    first_subblade_wrapper = last_subblade_wrapper = NULL;
+  } else {
+    if (!first_subblade_wrapper) return NULL;
+    blade = first_subblade_wrapper->blade_;
+  }
+
+  SubBladeWrapperWithList* ret = new SubBladeWrapperWithList(indices, count);
+
+  if (first_subblade_wrapper) {
+    ret->SetNext(last_subblade_wrapper);
+    first_subblade_wrapper->SetNext(ret);
+    last_subblade_wrapper = ret;
+  } else {
+    ret->SetNext(ret);
+    first_subblade_wrapper = last_subblade_wrapper = ret;
+  }
+
+  ret->SetupSubBlade(blade, 0, count);
```

Comment:
This will make primary() return true.
However, there is only supposed to be one primary subblade per string.
Currently, the "primary()" is based on whether the subblade has an offset of zero, which is why this will always return true right now.

We have two options:
1. We can modify this call so that the offset is only zero if the list of integers contains a zero.
2. We could add a primary_ bool and set it to true in SetNext() if next == this;

I'm not entirely certain which is better. The first one means fewer modifications to existing code, while the second one potentially fixes an existing problem.  (What if none of the subblades have offset == 0?)

The second is probably leaner code-wise, so maybe go with that?

Reply:
ok. How about now?